src/distribute_kv_cache/rpc/message.rs

```diff
@@ -0,0 +1,1985 @@
+use std::{fmt, mem};
+
+use async_trait::async_trait;
+use bytes::{Buf, BufMut, BytesMut};
+use clippy_utilities::{Cast, OverflowArithmetic};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use tracing::{debug, error, warn};
+
+use super::{
+    error::RpcError,
+    packet::{ActualSize, Decode, Encode, Packet},
+    utils::{get_u64_from_buf, u64_to_usize},
+};
+use crate::async_fuse::util::usize_to_u64;
+
+/// Zero copy conversion between num key and u8 buffers.
+#[allow(clippy::mem_forget)]
+#[allow(clippy::as_conversions)]
+#[allow(dead_code)]
+fn num_to_u8_buffer<K>(input: Vec<K>) -> Vec<u8>
+where
+    K: num::Num,
+{
+    let len = input.len() * mem::size_of::<K>();
+    let ptr = input.as_ptr();
+    let capacity = input.capacity();
+
+    std::mem::forget(input);
+
+    unsafe { Vec::from_raw_parts(ptr as *mut u8, len, capacity) }
```

Comment:
I think the capacity is not right. capacity means the number of elements that the vector can hold without reallocating. That means if you transfer the type of the vector, the capacity is different based on the type size in the vector.